package NQueryDataSaaS;

import "mapreduce/yt/interface/protos/extension.proto";

message TQDSaasErrorEntry {
    optional string SaasKey = 1             [(NYT.key_column_name) = "SaasKey"];
    optional string SaasPropName = 2        [(NYT.key_column_name) = "SaasPropName"];
    optional uint64 Timestamp = 4           [(NYT.key_column_name) = "Timestamp"];
    optional string Namespace = 5           [(NYT.column_name) = "Namespace"];
    optional string SourceFactorsHR = 6     [(NYT.column_name) = "SourceFactorsHR"];

    optional string Subkey_QueryStrong = 7  [(NYT.key_column_name) = "Subkey_QueryStrong"];
    optional string Subkey_QueryDoppel = 8  [(NYT.key_column_name) = "Subkey_QueryDoppel"];
    optional string Subkey_Url = 9          [(NYT.key_column_name) = "Subkey_Url"];
    optional string Subkey_Owner = 10       [(NYT.key_column_name) = "Subkey_Owner"];
    optional string Subkey_UrlMask = 11     [(NYT.key_column_name) = "Subkey_UrlMask"];
    optional string Subkey_ZDocId = 12      [(NYT.key_column_name) = "Subkey_ZDocId"];
    optional string Subkey_UserId = 13           [(NYT.key_column_name) = "Subkey_UserId"];
    optional string Subkey_UserLoginHash = 14    [(NYT.key_column_name) = "Subkey_UserLoginHash"];
    optional string Subkey_UserRegion = 15       [(NYT.key_column_name) = "Subkey_UserRegion"];
    optional string Subkey_UserRegionIpReg = 16  [(NYT.column_name) = "Subkey_UserRegionIpReg"];
    optional string Subkey_UserIpType = 17       [(NYT.column_name) = "Subkey_UserIpType"];
    optional string Subkey_StructKey = 18  [(NYT.key_column_name) = "Subkey_StructKey"];
    optional string Subkey_SerpTLD = 19     [(NYT.column_name) = "Subkey_SerpTLD"];
    optional string Subkey_SerpUIL = 20     [(NYT.column_name) = "Subkey_SerpUIL"];
    optional string Subkey_SerpDevice = 21  [(NYT.column_name) = "Subkey_SerpDevice"];
    optional uint32 SaasErrorCode = 22      [(NYT.column_name) = "SaasErrorCode"];
    optional string SaasErrorMessage = 23   [(NYT.column_name) = "SaasErrorMessage"];
    optional uint32 RetryCount = 24         [(NYT.column_name) = "RetryCount"];
    optional bool IsRemoval = 25            [(NYT.column_name) = "IsRemoval"];
    optional string Subkey_QueryDoppelToken = 26 [(NYT.key_column_name) = "Subkey_QueryDoppelToken"];
    optional string Subkey_QueryDoppelPair = 27  [(NYT.key_column_name) = "Subkey_QueryDoppelPair"];
    optional string Subkey_UrlNoCgi = 28  [(NYT.key_column_name) = "Subkey_UrlNoCgi"];
}
